> Perhaps that could be a good place to ensure that the desired behavior is captured.

In python/importlib_metadata@1a209e7e41, I started exploring adding a test that would capture the import time expectation (and at least indicate a regression when if the optimization was removed), but when I added the lazy import for csv, it detected no meaningful performance difference:

```
exercises.py:import time: 0:00:00.000001 (+0:00:00, 0%)
```

That indicates to me that the import time is 1 µs before and after the lazy import.

Attempting to understand why the timeit results weren't providing the expected performance gains, I added an assertion and found that the "setup" expression has a different outcome depending on what "test" expression is present:

```
 importlib_metadata feature/test-import-time @ .tox/py/bin/python -s -m timeit --setup 'import sys; assert "csv" not in sys.modules'
100000000 loops, best of 5: 3.9 nsec per loop
 importlib_metadata feature/test-import-time @ .tox/py/bin/python -s -m timeit --setup 'import sys; assert "csv" not in sys.modules' 'import importlib_metadata'
Traceback (most recent call last):
  File "/opt/homebrew/Cellar/python@3.12/3.12.1_1/Frameworks/Python.framework/Versions/3.12/lib/python3.12/timeit.py", line 330, in main
    number, _ = t.autorange(callback)
                ^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.12/3.12.1_1/Frameworks/Python.framework/Versions/3.12/lib/python3.12/timeit.py", line 226, in autorange
    time_taken = self.timeit(number)
                 ^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.12/3.12.1_1/Frameworks/Python.framework/Versions/3.12/lib/python3.12/timeit.py", line 180, in timeit
    timing = self.inner(it, self.timer)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<timeit-src>", line 3, in inner
    import sys; assert "csv" not in sys.modules
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError
```

That's weird, right? I would expect the "setup" to be run exactly once before any test expression was run, so it shouldn't be affected by the contents of the test expression, yet it is.

--- a/Lib/email/_parseaddr.py
+++ b/Lib/email/_parseaddr.py
@@ -194,6 +194,7 @@ def mktime_tz(data):
         # No zone info, so localtime is better assumption than GMT
         return time.mktime(data[:8] + (-1,))
     else:
+        # Delay the import, since mktime_tz is rarely used
         import calendar
 
         t = calendar.timegm(data)

--- a/Lib/importlib/metadata/__init__.py
+++ b/Lib/importlib/metadata/__init__.py
@@ -477,6 +477,8 @@ class Distribution(DeprecatedNonAbstract):
 
         @pass_none
         def make_files(lines):
+            # Delay csv import, since Distribution.files is not as widely used
+            # as other parts of importlib.metadata
             import csv
 
             return starmap(make_file, csv.reader(lines))

